iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -86,7 +88,7 @@ internal async Task<AmqpIoTSession> EnsureSessionAsync(TimeSpan timeout)
         {
             if (_closed)
             {
-                throw DEVICE_DISCONNECTED_EXCEPTION;
+                throw new IotHubException("AmqpUnit is disconnected.", false);
```

Comment:
The message of the exception needs to be clear and actionable for a developer or if exposed through UI. 
AmqpUnit is an internal type and messages of exceptions are part of the public API and should not change over time. Please change to a meaningful message.